File: SpitzeFront/src/models/Document.ts
export class Document {
   
        Id: String
        Timestamp: String
        TargetUrl: String
        RerferralUrl: String
        EntityModel: String
        Operation: OperationModel
        User: UserModel
        _rid: String
        _self: String
        _etag: String
        _attachments: String
        _ts: Number
      }
     
export class OperationModel{
    Type: String
      SearchResultCount: Number
      SearchTerm: String
}
export class UserModel{
      Id: Number
      Name: String
      Agent: String
}
export class TenantModel{
    Name: String
      TenantId: String
      ApiUser: String
}

File: SpitzeAPI/Models/Document.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace SpitzeAPI.Models
{
    [BsonIgnoreExtraElements]
    public class Document
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public string Id { get; set; }
        public string Timestamp { get; set; }
        public string TargetUrl { get; set; }
        public string RerferralUrl { get; set; }
        public EntityModel Entity { get; set; }
        public OperationModel Operation { get; set; }
        public UserModel User { get; set; }
        public TenantModel Tenant { get; set; }
        public string _rid { get; set; }
        public string _self { get; set; }
        public string _etag { get; set; }
        public string _attachments { get; set; }
        public long _ts { get; set; }
    }
}
